### PR TITLE
CPDRP-348: Import additional school emails from a CSV file

### PR DESCRIPTION
**Pull request by ethanmills**
### Context
CPDRP-348

Matt provided a CSV of emails scraped from school websites. This cleans and imports that list to a new table. When we come to import these emails in production we will transfer the CSV to the server via SCP and run the job there.

### Changes proposed in this pull request
- Create model for additional emails for a school
- Define a ruleset for cleaning additional emails
- Add a task to import emails from a CSV

### Testing
Unit testing, manually tested

### Review Checks
- [ ] All pages have automated accessibility checks via cypress
- [ ] All pages have visual tests via cypress + percy
- [x] All `School` queries are correctly scoped - `eligible` when they need to be

### How can I view this in a review app?
This can be tested as part of CPDRP-374

### CODE REVIEW

File: db/migrate/20210527113042_create_additional_school_emails.rb [hunk 1]
```diff
@@ -0,0 +1,12 @@
+# frozen_string_literal: true
+
+class CreateAdditionalSchoolEmails < ActiveRecord::Migration[6.1]
+  def change
+    create_table :additional_school_emails do |t|
+      t.references :school, null: false, foreign_key: true, type: :uuid
+      t.string :email, null: false
+
+      t.timestamps
+    end
```
Comment (tonyheadford) [inline db/migrate/20210527113042_create_additional_school_emails.rb:10]: Could we have a unique constraint here (email and school_id)?

**Review by tonyheadford — COMMENTED** [review]
Looks good, I think there's an opportunity to add a unique constraint in the migration and model. Also I think `email_address` is clearer than `email` (which to me is the communication) but not a blocker especially if this naming is already established.

File: app/services/additional_email_importer.rb [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+
+  def matches_keywords(email)
+    keywords = %w[head info enquiries office admin hello principal secretary reception]
+    keywords.any? { |keyword| email.include?(keyword) }
+  end
+
+  def matches_banned_words(email)
+    banned_words = %w[covid corona emergency admissions recruitment absences officer mailto]
+    banned_words.any? { |banned_word| email.include?(banned_word) }
+  end
```
Comment (kicferk1) [inline app/services/additional_email_importer.rb:56]: Soo, why do we need to do those checks exactly? Do we not trust the file we are getting?

Comment (kicferk1) [reply app/services/additional_email_importer.rb:56]: Oh, I guess it comes from scraping, so not really :P

Comment (ethanmills) [reply app/services/additional_email_importer.rb:56]: Definitely not...

**Review by kicferk1 — APPROVED** [review]

**Pull request merged**